ccan/ccan/tal/str/str.h

```diff
@@ -222,4 +222,14 @@ char *tal_strjoin_(const void *ctx,
 	tal_strreg_(ctx, string, TAL_LABEL(char, "[]"), __VA_ARGS__)
 bool tal_strreg_(const void *ctx, const char *string TAKES,
 		 const char *label, const char *regex, ...);
+
+/**
+ * tal_strlowering - match/extract from a string via (extended) regular expressions.
+ * @ctx: the context to tal from (often NULL)
+ * @string: the string to try to match (can be take())
+ */
+#define tal_strlowering(ctx, string)					\
+	tal_strlowering_(ctx, string)
+char *tal_strlowering_(const void *ctx, const char *string TAKES);
```

Comment:
This does not need to be macro-wrapped, and can simply go in common/utils.c for now (perhaps with a comment that it belongs in ccan).  We can promote it separately.